fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -287,6 +288,10 @@ public class Loan extends AbstractAuditableWithUTCDateTimeCustom {
     @Column(name = "loan_product_counter")
     private Integer loanProductCounter;
 
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "delinquency_range_id", nullable = true)
+    private DelinquencyRange delinquencyRange;
```

Comment:
@josehernandezfintecheandomx  We are storing the actual DeliquencyRange at 2 places:
- Loan.deliquencyRange
- LoanDeliquencyTags 

Please pick just one place to store the same data!
If you want this detail on the Loan, please rather add the LoanDeliquenctTag as association

Reply:
We need both:
`LoanDeliquencyTags` will be for the history (It is part of the requirement), and It could be used to know the latest tags, but How you will know when the Loan account has had one o more tags in the past but now It has not due It's not in arrears? So, that it the reason to use `Loan.deliquencyRange`

Reply:
@josehernandezfintecheandomx If LoanDeliquencyTags are for history:
-  please rename it to "LoanDeliquencyHistory"
- In the LoanDeliquencyTags, the loan OneToOne must be changed to ManyToOne as well!